dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/data/upload/RotatingDnsResolverTest.kt

```diff
@@ -111,6 +114,30 @@ internal class RotatingDnsResolverTest {
         assertThat(result2).containsExactlyElementsOf(fakeInetAddresses)
     }
 
+    @RepeatedTest(30)
+    fun `M not throw exception W concurrent access to lookup`(forge: Forge) {
+        // Given
+        // we need to keep the list of addresses low as it can only be reproduced with low number and it reflects
+        // the real use case where we have a small number of addresses to rotate
+        fakeInetAddresses = forge.aList(size = forge.anInt(min = 1, max = 3)) { mock() }
+        val countDownLatch = CountDownLatch(100)
+        whenever(mockDelegate.lookup(fakeHostname)) doReturn fakeInetAddresses
+        // just wait the TTL time to make sure all threads are concurrently accessing the lookup
+        Thread.sleep(TEST_TTL_MS.inWholeMilliseconds)
+
+        // When
+        repeat(100) {
+            Thread {
+                assertThatNoException().isThrownBy {
+                    Thread.sleep(forge.aLong(min = 0, max = 100))
+                    testedDns.lookup(fakeHostname)
+                }
+                countDownLatch.countDown()
+            }.start()
```

Comment:
if I'm not wrong, this will never throw in the end, the thread spinned will be killed, but error won't be propagated further and won't stop the test.

Moreover, since `countDownLatch.countDown()` won't happen in case of the error, `countDownLatch.await()` will be waiting forever.

Reply:
hmmm not sure based on my tests, theoretically `assertThatNoException` should stop the test with a fail. In that case the `countDownLatch` will not wait anymore. I will double check this though.